modules/material/material.service.js

```diff
@@ -17,21 +18,25 @@ class MaterialsService {
   }
 
   async addMaterial(data) {
+    if (await this.checkMaterialExist(data)) {
+      throw new Error(MATERIAL_ALREADY_EXIST);
+    }
+    return new Material(data).save();
+  }
+
+  async deleteMaterial(id) {
+    return await Material.findByIdAndDelete(id);
+  }
+
+  async checkMaterialExist(data) {
     const material = await Material.find({
```

Comment:
Use the method `count` here to reduce the amount of data that is transferred between app and db